src/python/pants/backend/python/providers/pyenv/rules_integration_test.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import shutil
+from textwrap import dedent
+from typing import Tuple
+
+import pytest
+
+from pants.backend.python import target_types_rules
+from pants.backend.python.dependency_inference import rules as dependency_inference_rules
+from pants.backend.python.goals.run_python_source import PythonSourceFieldSet
+from pants.backend.python.goals.run_python_source import rules as run_rules
+from pants.backend.python.providers.pyenv.rules import rules as pyenv_rules
+from pants.backend.python.target_types import PythonSourcesGeneratorTarget
+from pants.build_graph.address import Address
+from pants.core.goals.run import RunRequest
+from pants.engine.process import InteractiveProcess
+from pants.engine.rules import QueryRule
+from pants.engine.target import Target
+from pants.testutil.rule_runner import RuleRunner, mock_console
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *run_rules(),
+            *pyenv_rules(),
+            *dependency_inference_rules.rules(),
+            *target_types_rules.rules(),
+            QueryRule(RunRequest, (PythonSourceFieldSet,)),
+        ],
+        target_types=[
+            PythonSourcesGeneratorTarget,
+        ],
+    )
+
+
+def run_run_request(
+    rule_runner: RuleRunner,
+    target: Target,
+) -> Tuple[int, str, str]:
```

Comment:
using a `namedtuple` would make this less convoluted on the receiving end.

Reply:
I stole this from the run tests 🤷‍♂️